### PR TITLE
Setup the bugziller VM

### PR DESCRIPTION
**Pull request by nigelbabu**

### CODE REVIEW

File: hosts [hunk 1]
```diff
@@ -95,10 +95,10 @@ builder7.rht.gluster.org
 builder8.rht.gluster.org
 builder9.rht.gluster.org
 
-
 [jenkins_builders:children]
 jenkins_builders_rht
 jenkins_builders_rackspace
+bugziller
```
Comment (mscherer) [inline hosts]: This mean this is gonna setup a whole jenkins builder env, ie, everything to test glusterfs, is that also what we want ?

File: playbooks/deploy_bugziller.yml [hunk 2]
```diff
@@ -1,20 +1,3 @@
...
-    bridge: virbr1
-    sshkey: "{{ lookup('file', '~/.ssh/id_rsa.pub') }}"
-    network:
-      bootproto: static
-      ip: 8.43.85.178/24
-      gateway: 8.43.85.254
-      nameservers:
-      - 8.8.8.8
-      - 8.8.4.4
+  - role: bugzilla
```
Comment (mscherer) [inline playbooks/deploy_bugziller.yml]: Shouldn't it be called "bugziller", rather than "bugzilla" ? It doesn't install bugzilla.

File: roles/jenkins_builder/tasks/main.yml [hunk 3]
```diff
@@ -63,6 +63,7 @@
   - /d/build
   - /d/logs
   - /home/jenkins/root
+  - /home/jenkins/root/workspace
```
Comment (mscherer) [inline roles/jenkins_builder/tasks/main.yml]: This seems unrelated to bugziller deployment, what prompted this ?

Comment (nigelbabu) [inline roles/jenkins_builder/tasks/main.yml]: I actually don't remember.

File: hosts (hunk 1 above)
Comment (nigelbabu) [inline hosts]: We have two options. Trim down what a base jenkins builder env means. Or not care. I'm going with not care for now :)

I need some of the things, but not all of them.

File: roles/jenkins_builder/tasks/main.yml (hunk 3 above)
Comment (nigelbabu) [reply roles/jenkins_builder/tasks/main.yml]: Without this folder being precreated the setfacl command in the next step will fail on fresh nodes

**Comment by mscherer** [comment]
So I may have changed a big the existing stuff, I think you can drop   playbooks/deploy_bugziller_vm.yml

File: roles/bugziller/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+---
+- name: Install python-bugzilla
+  package: name=python-bugzilla state=latest
+
+- name: Setup python-bugzilla
+  command: "bugzilla login {{bugzilla_user}} {{bugzilla_pass}}"
```
Comment (mscherer) [inline roles/bugziller/tasks/main.yml]: We need to have them in the private repo to work.

Also, how do we handle 1) cookie expiration 2) password change ?

(2 is less critical, but 1 bother me a bit)

Also, do we need a specific doc for that, or we can assume that "this is a service account, follow usual bugzilla process for any change aka send mail to xxx" is enough ?

File: roles/bugziller/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+  args:
+    creates: "/home/jenkins/.bugzillacookies"
+  when: bugzilla_user is defined
+  no_log: True
+  become: yes
+  become_user: jenkins
```
Comment (mscherer) [inline roles/bugziller/tasks/main.yml]: So we need to have the jenkins env being setup, so this should be a dependency, which in turn mean we can drop adding it to jenkins_nodes:children ?

File: roles/bugziller/tasks/main.yml (hunk 4 above)
Comment (nigelbabu) [reply roles/bugziller/tasks/main.yml]: It's already in the private repo, so we have that sorted. Don't we keep running ansible on all hosts every day?

File: roles/bugziller/tasks/main.yml (hunk 5 above)
Comment (nigelbabu) [reply roles/bugziller/tasks/main.yml]: Yeah, we need a subset of the Jenkins environment setup. So we might as well do that.

File: roles/bugziller/tasks/main.yml (hunk 4 above)
Comment (nigelbabu) [reply roles/bugziller/tasks/main.yml]: (Ah, it's not in the private repo. The change is ready but not pushed)

File: roles/jenkins_builder/tasks/main.yml (hunk 3 above)
Comment (mscherer) [reply roles/jenkins_builder/tasks/main.yml]: Indeed, and it seems to be related to the failure we see on https://bugzilla.redhat.com/show_bug.cgi?id=1488125 so I cherry picked that fix.

File: roles/bugziller/tasks/main.yml (hunk 4 above)
Comment (mscherer) [reply roles/bugziller/tasks/main.yml]: Indeed, that's run nightly so this would work. For some reason, I tought that the bugzilla login command was run just if we had no cookie file, but since we run it each time, i guess that's ok.

Now, if bugzilla login do not create a cookie if one already exist, then we will have a problem.

**Comment by nigelbabu** [comment]
@mscherer Can you take a look at this again?

File: playbooks/deploy_bugziller.yml [hunk 6]
```diff
@@ -0,0 +1,3 @@
+- hosts: bugziller.rht.gluster.org
```
Comment (mscherer) [inline playbooks/deploy_bugziller.yml]: Wrong hostname, the vm is now bugzilla.int.rht.gluster.org

**Comment by mscherer** [comment]
Done in another commit
